produce/lib/produce/options.rb

```diff
@@ -137,19 +137,7 @@ def self.available_options
                                      default_value: CredentialsManager::AppfileConfig.try_fetch_value(:itc_team_name),
                                      verify_block: proc do |value|
                                        ENV["FASTLANE_ITC_TEAM_NAME"] = value.to_s
-                                     end),
-        FastlaneCore::ConfigItem.new(key: :merchant_name,
-                                      short_option: "-r",
-                                      env_name: "PRODUCE_MERCHANT_NAME",
-                                      description: "Name for the merchant that is created",
-                                      optional: true),
-        FastlaneCore::ConfigItem.new(key: :merchant_identifier,
-                                       short_option: "-o",
-                                       env_name: "PRODUCE_MERCHANT_IDENTIFIER",
-                                       description: "Merchant identifier for the merchant",
-                                       verify_block: proc do |value|
-                                         UI.user_error!("Merchant ID identifiers should start with 'merchant.'") unless value.start_with?("merchant.")
-                                       end)
```

Comment:
@ohayon These are removed following @KrauseFx's comment above about the new options appearing when they are not relevant when using `bundle exec produce --help`. They are not options for the other commands such as `create` and `group` and should not be included. With the latest changes this no longer happens yet they will still appear in help for the specific merchant command, eg `bundle exec fastlane produce merchant --help`